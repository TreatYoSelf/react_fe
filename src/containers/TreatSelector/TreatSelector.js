import React from 'react';
import { StyleSheet, Text, Image, TouchableOpacity } from 'react-native';
import * as All from '../../helpers/assets';

export default function TreatSelector({ id, title, selectCategory, style }) {
    return (
        <TouchableOpacity style={[styles.container, style]} onPress={() => selectCategory(`${title}`, id)}>
            <Image source={All[`${title.toLowerCase()}`]} style={{ width: 50, height: 50 }} />
            <Text style={styles.header}>{title}</Text>
        </TouchableOpacity>
    );
}

const styles = StyleSheet.create({
    container: {
        alignItems: 'center',
        justifyContent: 'space-around',
        flexDirection: 'row',
        width: 300,
        padding: 10,
        backgroundColor: "#e6f7ff",
        borderWidth: 5,
        borderColor: '#003045'
    },
    header: {
        fontWeight: 'bold'
    },
    selected: {
        backgroundColor: "blue",
    }
});