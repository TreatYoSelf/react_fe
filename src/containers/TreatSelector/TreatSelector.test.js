import React from 'react';
import renderer, { act } from 'react-test-renderer';

import TreatSelector from './TreatSelector';

describe('<TreatSelector />', () => {
    it('renders correctly', async () => {
        let tree;
        tree = renderer.create(<TreatSelector />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});