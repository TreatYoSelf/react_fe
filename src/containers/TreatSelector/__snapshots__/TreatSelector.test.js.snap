// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TreatSelector /> renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#e6f7ff",
      "borderColor": "#003045",
      "borderWidth": 3,
      "flexDirection": "row",
      "justifyContent": "space-around",
      "opacity": 1,
      "padding": 10,
    }
  }
>
  <Image
    style={
      Object {
        "height": 50,
        "width": 50,
      }
    }
  />
  <Text
    style={
      Object {
        "fontWeight": "bold",
      }
    }
  >
    testTreat
  </Text>
</View>
`;
